test/blackbox-tests/test-cases/runtest-cmd-tests.t

```diff
@@ -0,0 +1,99 @@
+Test running tests by specifying ML source files directly.
+
+  $ cat > dune-project <<EOF
+  > (lang dune 3.21)
+  > EOF
+
+Set up a simple test with multiple test executables:
+
+  $ cat > dune <<EOF
+  > (tests
+  >  (names test1 test2)
+  >  (modules test1 test2 test_lib))
+  > EOF
+
+  $ cat > test_lib.ml <<EOF
+  > let run_test name =
+  >   Printf.printf "Running %s\n" name
+  > EOF
+
+  $ cat > test1.ml <<EOF
+  > let () =
+  >   Test_lib.run_test "test1";
+  >   assert true
+  > EOF
+
+  $ cat > test2.ml <<EOF
+  > let () =
+  >   Test_lib.run_test "test2";
+  >   assert false (* This test fails *)
+  > EOF
+
+Running tests by specifying ML files directly:
+
+  $ dune test test1.ml
+  Running test1
+
+  $ dune test test2.ml
+  File "dune", line 2, characters 14-19:
+  2 |  (names test1 test2)
+                    ^^^^^
+  Running test2
+  Fatal error: exception Assert_failure("test2.ml", 3, 2)
+  [1]
+
+Error when specifying a non-test ML file:
+
+  $ cat > lib.ml <<EOF
+  > let x = 42
+  > EOF
+
+  $ cat > dune <<EOF
+  > (library
+  >  (name mylib)
+  >  (modules lib))
+  > (tests
+  >  (names test1 test2)
+  >  (modules test1 test2 test_lib))
+  > EOF
+
+  $ dune test lib.ml
+  Error: "lib.ml" does not match any known test.
+  [1]
+
+Error when specifying a non-existent ML file:
+
+  $ dune test nonexistent.ml
+  Error: "nonexistent.ml" does not match any known test.
+  [1]
+
+  $ dune test test3.ml
+  Error: "test3.ml" does not match any known test.
+  Hint: did you mean test1.ml or test2.ml?
+  [1]
+
+Can run tests from _build directory:
+
+  $ rm -rf _build
+  $ dune test _build/default/test1.ml
+  Running test1
+
+Can specify multiple ML files:
+
+  $ rm -rf _build
+  $ dune test test1.ml test2.ml
+  Running test1
+  File "dune", line 5, characters 14-19:
+  5 |  (names test1 test2)
+                    ^^^^^
+  Running test2
+  Fatal error: exception Assert_failure("test2.ml", 3, 2)
+  [1]
+
+When specifying a library module that is part of a test stanza, run all tests
```

Comment:
I think this behavior is weird. There should be one name for a test and not multiple.